src/data-model-providers/codegen/ServerClusterInterfaceRegistry.h

```diff
@@ -31,13 +31,17 @@ namespace app {
 struct ServerClusterRegistration
 {
     // A single-linked list of clusters registered for the given `endpointId`
-    ServerClusterInterface * serverClusterInterface = nullptr;
-    ServerClusterRegistration * next                = nullptr;
+    ServerClusterInterface * const serverClusterInterface;
+    ServerClusterRegistration * next;
 
-    constexpr ServerClusterRegistration() = default;
-    constexpr ServerClusterRegistration(ServerClusterInterface * interface, ServerClusterRegistration * next_item = nullptr) :
-        serverClusterInterface(interface), next(next_item)
+    constexpr ServerClusterRegistration(ServerClusterInterface & interface, ServerClusterRegistration * next_item = nullptr) :
+        serverClusterInterface(&interface), next(next_item)
```

Comment:
Fwiw, our member could become a reference too... Followup for that, please.

Reply:
created #37909